graph-api.yaml

```diff
@@ -0,0 +1,459 @@
+    post:
+      summary: Validate RDF data against SHACL shapes
+      description: |
+        Submit RDF data and SHACL shapes to validate the data. 
+        The server processes the request and returns a SHACL validation report.
+      parameters:
+        - $ref: '#/components/parameters/dataset'
+      requestBody:
+        description: RDF data and SHACL shapes to validate.
+        required: true
+        content:
+          application/ld+json:
+            schema:
+              type: object
+              properties:
+                data:
+                  type: string
+                  description: RDF data to validate.
+                  format: string
+                shapes:
+                  type: string
+                  description: SHACL shapes to validate the RDF data against.
+                  format: string
+              required:
+                - data
+                - shapes
+          application/turtle:
+            schema:
+              type: string
+              description: RDF data and SHACL shapes in Turtle format.
+          application/n-triples:
+            schema:
+              type: string
+              description: RDF data and SHACL shapes in N-Triples format.
+      responses:
+        200:
+          description: Validation report generated successfully.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ValidationReport'
+            application/ld+json:
+              schema:
+                $ref: '#/components/schemas/ValidationReport'
+        400:
+          description: Invalid input data or SHACL shapes.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        500:
+          description: Internal Server Error.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+
+  /{dataset}/graphql:
+    get:
+      summary: Execute a GraphQL query against the Dataset schema
+      parameters:
+        - $ref: '#/components/parameters/dataset'
+        - $ref: '#/components/parameters/graphQLQuery'
+        - $ref: '#/components/parameters/operationName'
+        - $ref: '#/components/parameters/variables'
+        - $ref: '#/components/parameters/extensions'
+      responses:
+        '200':
+          $ref: '#/components/responses/GraphQLSuccess'
+        '400':
+          $ref: '#/components/responses/GraphQLError'
+    post:
+      summary: Execute a GraphQL query against the Dataset schema (POST)
+      parameters:
+        - $ref: '#/components/parameters/dataset'
+      requestBody:
+        $ref: '#/components/requestBodies/GraphQLRequest'
+      responses:
+        '200':
+          $ref: '#/components/responses/GraphQLSuccess'
+        '400':
+          $ref: '#/components/responses/GraphQLError'
+
+  /{dataset}/traversal/graphql:
```

Comment:
Fuseki doesn't allow more than one `/` after the `/{dataset}/` portion so think we can remove this endpoint from the definition

There's a number of other endpoint definitions with similar issues

Reply:
That's interesting. Hadn't observed that and was about to challenge you when I realised my previous queries (like below) were against GraphQL Jena.  

`curl --location 'http://localhost:11666/dataset/traversal/graphql' \
--header 'Content-Type: application/json' \
--data '{"query":"query Nodes($uris: [String!]!) {\n    nodes(uris: $uris) {\n      id\n      uri\n      uriHash\n      shortUri\n      types {\n        shortUri\n        uri\n      }\n      properties {\n        shortPredicate\n        value\n      }\n      inRels {\n        id\n        domain_id\n        predicate\n        range_id\n      }\n      outRels {\n        id\n        range_id\n        domain_id\n        predicate\n      }\n    }\n  }","variables":{"uris":["https://starwars.com#person_HanSolo","https://starwars.com#person_KyloRen"]}}'`

For my own interest, how do we differentiate between the GraphQL schemas to be used in our queries via SC Graph?